packages/execution-environments/src/common/validation.ts

```diff
@@ -0,0 +1,40 @@
+import { HandlerType } from '@metamask/snap-utils';
+
+const VALIDATION_FUNCTIONS = {
+  [HandlerType.OnRpcRequest]: validateFunctionExport,
+  [HandlerType.OnTransaction]: validateFunctionExport,
+  [HandlerType.SnapKeyring]: validateKeyringExport,
+};
+
+/**
+ * Validates a function export.
+ *
+ * @param snapExport - The export itself.
+ * @returns True if the export matches the expected shape, false otherwise.
+ */
+function validateFunctionExport(snapExport: any) {
+  return typeof snapExport === 'function';
+}
+
+/**
+ * Validates a keyring export.
+ *
+ * @param snapExport - The export itself.
+ * @returns True if the export matches the expected shape, false otherwise.
+ */
+function validateKeyringExport(snapExport: any) {
```

Comment:
```suggestion
function validateKeyringExport(snapExport: unknown): snapExport is typeof Keyring {
```
Or something like this?